src/js/App/Sidenav/Navigation/DynamicNav.js

```diff
@@ -0,0 +1,44 @@
+import React, { useState, useEffect } from 'react';
+import { useModule } from '@scalprum/react-core';
+import PropTypes from 'prop-types';
+import { Skeleton } from '@patternfly/react-core';
+import ChromeNavItemFactory from './ChromeNavItemFactory';
+import { useHistory } from 'react-router';
+import { useDispatch, useSelector } from 'react-redux';
+import { loadLeftNavSegment } from '../../../redux/actions';
+
+const DynamicNav = ({ dynamicNav, ...rest }) => {
+  const [nav, setNav] = useState();
+  const {
+    location: { pathname },
+  } = useHistory();
+  const dispatch = useDispatch();
+  const [appName] = dynamicNav.split('/');
+  const currentNamespace = pathname.split('/')[1];
+  const schema = useSelector(({ chrome: { navigation } }) => navigation[currentNamespace]);
+  const { default: navigation } = useModule(appName, './Navigation', {});
```

Comment:
Is the empty object argument required?

Reply:
Yup, the `useModule` uses state and when we use the destruct here `default: navigation` it would throw an error of destructing `undefined`